design/ceph-csi-driver.md

```diff
@@ -54,6 +56,11 @@ If CSI drivers are already successfully deployed, Rook should help monitor the S
 
 ### Rook initiated CSI drivers deployment
 
-If Rook is instructed to deploy CSI drivers and other external controllers, Rook Operator should create all the driver and daemon deployments as well.
+If Rook is instructed to deploy CSI drivers and other external controllers, Rook Operator should create all the driver and daemon deployments as well. Since there are breaking changes in CSI 1.0 spec and Kubernetes 1.13, it makes sense to focus on newer CSI specs and only on latest Kubernetes releases (i.e. 1.13 and beyond).
+
+The projected changes to Rook include the following:
+
+- Vendor in Ceph CSI drivers and create CSI driver binaries in Rook Ceph container image, rather than directly consuming ceph-csi images.
```

Comment:
Just wondering, is there a reason why Rook needs to vendor the drivers in and not just the `ceph-csi` image(s)?

Reply:
For now, we don't have resource to run ceph-csi on multiple kubernetes and ceph releases. Vendoring it ceph-csi pkg and build drivers with the same rook ceph container image allows us to verify against the same kubernetes and ceph release that rook supports.